zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java

```diff
@@ -138,11 +146,131 @@ public static Either<ProblemDetail, ProcessInstanceSort> toSearchQuerySort(
+          builder.asc();
+        } else if ("desc".equalsIgnoreCase(order)) {
+          builder.desc();
+        } else {
+          throw new RuntimeException("unkown sortOrder " + order);
+        }
+      }
+
+      return Either.right(builder.build());
+    }
+
+    return Either.right(null);
+  }
+
+  public static Either<ProblemDetail, UserTaskFilter> toUserTaskFilter(
+      final UserTaskFilterRequest filter) {
+    final var builder = FilterBuilders.userTask();
+
+    if (filter != null) {
+      // key
+      if (filter.getKey() != null) {
+        builder.userTaskKeys(filter.getKey());
+      }
+
+      // state
+      if (filter.getTaskState() != null && !filter.getTaskState().isEmpty()) {
+        builder.userTaskState(filter.getTaskState());
+      }
+
+      // assignee
+      if (filter.getAssignee() != null && !filter.getAssignee().isEmpty()) {
+        builder.assignees(filter.getAssignee());
+      }
+
+      // candidateGroup
+      if (filter.getCandidateGroup() != null && !filter.getCandidateGroup().isEmpty()) {
+        builder.candidateGroups(filter.getCandidateGroup());
+      }
+
+      // candidateUser
+      if (filter.getCandidateUser() != null && !filter.getCandidateUser().isEmpty()) {
+        builder.candidateUsers(filter.getCandidateUser());
+      }
+
+      // processDefinitionKey
+      if (filter.getProcessDefinitionKey() != null) {
+        builder.processDefinitionKeys(filter.getProcessDefinitionKey().toString());
+      }
+
+      // processInstanceKey
+      if (filter.getProcessInstanceKey() != null) {
+        builder.processInstanceKeys(filter.getProcessInstanceKey());
+      }
+
+      // tenantIds
+      if (filter.getTenantIds() != null) {
+        builder.tenantIds(filter.getTenantIds());
+      }
+
+      // creationTime
+      if (filter.getCreationTime() != null) {
+        builder.creationDate(toDateValueFilter(filter.getCreationTime()));
+      }
+
+      // completionTime
+      if (filter.getCompletionTime() != null) {
+        builder.completionDate(toDateValueFilter(filter.getCompletionTime()));
+      }
+
+      // dueDate
+      if (filter.getDueDate() != null) {
+        builder.dueDate(toDateValueFilter(filter.getDueDate()));
+      }
+
+      // followUpDate
+      if (filter.getFollowUpDate() != null) {
+        builder.followUpDate(toDateValueFilter(filter.getFollowUpDate()));
+      }
+    }
+
+    return Either.right(builder.build());
+  }
+
+  public static Either<ProblemDetail, UserTaskQuery> toUserTaskQuery(
+      final UserTaskSearchQueryRequest request) {
+    final var page = toSearchQueryPage(request.getPage()).get();
+    final var sorting = toUserTaskSearchQuerySort(request.getSort()).get();
+    final var userTaskFilter = toUserTaskFilter(request.getFilter()).get();
+    return Either.right(
+        SearchQueryBuilders.userTaskSearchQuery()
+            .page(page)
+            .filter(userTaskFilter)
+            .sort(sorting)
+            .build());
```

Comment:
🔧 If any of those mapping methods can actually return an `Either.Left`, you should also return that `Left` here and not create the `Right` type. Otherwise, you'll suppress the problem detail created by any of those methods.